src/components/rmrk/Gallery/CollectionItem.vue

```diff
@@ -244,5 +280,31 @@ export default class CollectionItem extends Vue {
   get iframeSettings(): Record<string, unknown> {
     return { width: '100%', height: '100vh' }
   }
+
+  protected listEventsBeforeTime(nftEvents: Interaction[][], time: string) {
+    return nftEvents.map(events => {
+      let properEvents = events.filter((e: any) => !isAfter(new Date(e.timestamp), new Date(time)))
+      return properEvents.length && properEvents[properEvents.length - 1].interaction === 'LIST' ? [properEvents[properEvents.length - 1]] : []
+    })
+  }
+
+  protected onlyEvents(nft:NFT){
+    return nft.events.map((e:Interaction) => {
+      return {...e, name: nft.name}
+    })
+  }
+  protected priceEvents(nftEvents:Interaction[]){
+    return nftEvents.filter((event:Interaction) => event.interaction === 'LIST' || event.interaction === 'BUY' || event.interaction === 'SEND' || event.interaction === 'CONSUME')
+  }
+  protected onlyBuyEvents(nftEvents:any[]){
```

Comment:
this is already in utils

Reply:
Can you point where save function in utils?
Thanks!

Reply:
@devVenus1202 search for 
```js 
function pairListBuyEvent
```

Reply:
still unresolved

Reply:
so the logic behind is that current daily floor is calculated as:

```ts
Math.min(...this.nfts.map(nft => Number(nft.price)).filter(price => price > 0))
```

but this won't work because current state automatically filters out burned.

For the floor per day: We need to check other events because as @devVenus1202 pointed out it changes the floor price.

But still I would omit the usage of three or `||` in the condition.

Third: Please write a test for this